enumerables.rb

```diff
@@ -0,0 +1,141 @@
+require 'byebug'
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    if is_a? Array
+      while i <= length - 1
+        yield (self[i])
+        i += 1
+      end
+    elsif is_a? Hash
+      arr = to_a
+      while i <= length - 1
+        yield [arr[i][0], arr[i][1]]
+        i += 1
+      end
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    if is_a? Array
+      while i <= length - 1
+        yield [self[i], i]
+        i += 1
+      end
+    elsif is_a? Hash
+      arr = to_a
+      while i <= length - 1
+        yield [arr[i], i]
+        i += 1
+      end
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a? Hash
+      hashy = {}
+      my_each { |item| hashy[item.to_a[0]] = item.to_a[1] if yield i.to_a[0], i.to_a[1] }
+    else
+      arr = []
+      to_a.my_each { |item| arr << item if yield(item) }
+      arr
+    end
+  end
+
+  def my_all?(data = nil)
+    return my_none?(data) if block_given? && !data.nil?
+
+    arr = to_a
+    return true if arr.empty?
+
+    if block_given?
+      to_a.my_each { |item| return false unless yield(item) }
+    elsif data.nil? && !block_given?
+      to_a.my_each { |item| return false unless item }
+    elsif data.is_a? Class
+      to_a.my_each { |item| return false unless item.is_a? data }
+    elsif data.is_a? Regexp
+      to_a.my_each { |item| return false unless item.to_s.match(data) }
+    end
+    true
+  end
```

Comment:
- For my_all? handle the case when a pattern other than Regex or a Class is given, it should return true if all of the collection matches the pattern.
 
For example
The built-in method
`[2,3,4,5,6].all?(3)` returns `false`


Your method
`[2,3,4,5,6].my_all?(3)` returns `true`

- It should return false since not all elements in the array are 3